lib/vsc/administration/slurm/sync.py

```diff
@@ -342,6 +571,63 @@ def slurm_vo_accounts(account_page_vos, slurm_account_info, clusters, host_insti
     return commands
 
 
+def slurm_project_users_accounts(project_members, active_accounts, slurm_user_info, clusters):
+    """Check if the users are in the project account.
+
+    For users in the project:
+
+    - If the user does not exist on the system:
+        - a new association in the default account is created. The associated QoS does not allow jobs.
+        - a new association in the project account is created.
+
+    - For users who left the project:
+        - The user's association in the project is removed.
+        - If this was the last project for this user, we also remove the association in the default account.  (TODO)
+    """
+
+    commands = []
+
+    for cluster in clusters:
+        cluster_users_acct = [
+            (user.User, user.Account) for user in slurm_user_info if user and user.Cluster == cluster
+        ]
+
+        new_users = set()
+        remove_project_users = set()
+        all_project_users = set()
+
+        for (members, project_name) in project_members:
+
+            # these are the current Slurm users for this project
+            slurm_project_users = set([user for (user, acct) in cluster_users_acct if acct == project_name])
+            all_project_users |= slurm_project_users
+
+            # these users are not yet in the Slurm DBD for any project
+            new_users |= set([(user, project_name) for user in (members & active_accounts) - slurm_project_users])
+
+            # these are the Slurm users that should no longer be associated with the project
+            remove_project_users |= set([(user, project_name) for user in slurm_project_users - members])
+
+        # these are the users not in any project
+        remove_slurm_users = set([u[0] for u in cluster_users_acct]) - all_project_users
```

Comment:
nothing is done iwth this list. if this is not empty, i would assume and error/failure?

Reply:
I have not decided what to do with these. I would imagine that they should be removed from the system. But maybe fail hard if this list is not empty and trigger an alert.

Reply:
Those are the gadminforever users and vsc40003. I assume you do not want these removed :p

Reply:
the ones that you know will end up in here should not logged ofcourse; but all the other should. but i don;t understand why these are not in a project also